accord-core/src/main/java/accord/primitives/AbstractRanges.java

```diff
@@ -56,7 +56,10 @@ public int indexOf(RoutableKey key)
     @Override
     public int indexOf(Range find)
```

Comment:
I think we should rename `indexOf` to `ceilIndexOf` just so callers are aware it's not necessarily unique. Or have it require the `Search` parameter.

Reply:
I am not a fan of this as this method is coming from `accord.primitives.Routables#indexOf` and that doesn't really make sense to me with the other 2 implementations (finding a key in a set of keys).

One possibility is to add a *new* method on `AbstractRanges` that lets you define the `Search` param and have `indexOf` return the *first* match.  The only code path I see that might use this new API is `accord.primitives.AbstractRanges#intersects(accord.primitives.Range)` as it could possible do less checks as it can return after the first match

Reply:
I would prefer to remove `indexOf` from `Routables` and introduce a new `indexOf` method only for `RoutableKey` types (e.g. RoutingKeys and Keys). We shouldn't really _have_ an `indexOf` method that accepts a `Range` for `AbstractRanges` as its poorly defined.

Reply:
pushed.  `indexOf` is now in `AbstractKeys` and `AbstractRanges` but they don't share the same signature; ranges asks for a search op.  I added a new `firstIndexOf` that uses `CEIL` with the new method in the single place that called this before.

Reply:
I think it would be cleaner to either call it `ceilIndexOf` or just supply/accept `CEIL` parameter, so we don't mix descriptors of functionality.

Reply:
I used `firstIndexOf` to be similar to `java.util.List#lastIndexOf`, but switched to `ceilIndexOf` for now